alarm-datastore-cassandra/src/main/java/com/telecominfraproject/wlan/alarm/datastore/cassandra/AlarmDatastoreCassandra.java

```diff
@@ -716,11 +757,20 @@ public void resetAlarmCounters() {
             strb.append(") ");
 
             query += strb.toString();
+            filterOptions = FilterOptions.customer_and_alarmCode;
         }        
 
         if(createdAfterTimestamp > 0) {
         	query += " and createdTimestamp > ?" ;
         	queryArgs.add(createdAfterTimestamp);
+        	filterOptions = FilterOptions.customer_and_timestamp;
+        }
+        
+        if (acknowledged != null) {
+        	query_head = "select customerId, equipmentId, alarmCode, createdTimestamp from alarm_by_acknowledged where customerId = ? ";
+        	query += " and acknowledged = ? ";
+        	queryArgs.add(acknowledged);
```

Comment:
nope. queryArgs by this point may contain whole bunch of other stuff from other parameters. need to clear them and add customerId explicitly. also please add a test that covers this scenario.

Reply:
Fixed in latest commit 
Added test cases for a mix of filters that include acknowledged flag